src/test/java/io/r2dbc/mssql/util/FluxDiscardOnCancelUnitTests.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.r2dbc.mssql.util;
+
+import org.junit.jupiter.api.Test;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Hooks;
+import reactor.test.StepVerifier;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.stream.IntStream;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * Unit tests for {@link FluxDiscardOnCancel}.
+ *
+ * @author Mark Paluch
+ */
+class FluxDiscardOnCancelUnitTests {
+
+    @Test
+    void shouldEmitAllItemsOnSubscription() {
+
+        Iterator<Integer> items = createItems(4);
+
+        Flux.fromIterable(() -> items)
+            .as(Operators::discardOnCancel)
```

Comment:
take a look at `Flux#transform` (and probably `Operators.lift`), `as` does not apply `onAssembly` hook or anything else